hypertrace-core-graphql-trace-schema/src/main/java/org/hypertrace/core/graphql/trace/dao/GatewayServiceTraceDao.java

```diff
@@ -24,6 +33,9 @@
   private final GraphQlGrpcContextBuilder grpcContextBuilder;
   private final GatewayServiceTraceRequestBuilder requestBuilder;
   private final GatewayServiceTraceConverter traceConverter;
+  private static final String TRACE_FETCH_LATENCY = "graphql.trace.fetch.latency";
+  private final ConcurrentMap<String, Timer> tenantToTraceFetchTime = new ConcurrentHashMap<>();
```

Comment:
Why are we keying this off tenant? What does the tenant have to do with the metric other than being an extra point of metadata? Do we even need to hold past requests?

Reply:
@aaron-steinfeld Basically, this is trying to maintain a separate Timer for each tenant so that we can report tenant id as a metric tag.
The Timer instance would be continuously flushing the metrics so this shouldn't hold the past requests.

Reply:
I find the API strange I guess - I'd expect a single timer (in other words, a client) that we can add attributes on to measurements. Putting the burden on the caller to manage the objects based on the metadata they want to apply seems like unnecessary overhead that we could hopefully cut out.